Update s3 gateway docs with recent changes (#4088)
pachyderm/pachyderm: doc/enterprise/s3gateway.md

access, use the PFS gRPC interface instead, or one of the
[client drivers](https://github.com/pachyderm/python-pachyderm).

## Authentication

If auth is enabled on the Pachyderm cluster, credentials must be passed with
each s3 gateway endpoint using AWS' signature v2 or v4 methods. Object store
tools and libraries provide built-in support for these methods, but they do
not work in the browser. When you use authentication, set the access and secret key
to the same value; they are both the Pachyderm auth token used
to issue the relevant PFS calls.

If auth is not enabled on the Pachyderm cluster, no credentials need to be
passed to s3gateway requests.

## Buckets

The S3 gateway presents each branch from every Pachyderm repository as
an S3 bucket.
For example, if you have a `master` branch in the `images` repository,
an S3 tool sees `images@master` as the `master.images` S3 bucket.

## Versioning

Most operations act on the HEAD of the given branch. However, if your object
store library or tool supports versioning, you can get objects in non-HEAD
commits by using the commit ID as the version.

## Port Forwarding

> If you do not have direct access to the kubernetes cluster, you can use port

Reviewer: ```suggestion
If you do not have direct access to the Kubernetes cluster, you can use port
```

forwarding instead. Simply run `pachctl port-forward`, which will allow you
to access the s3 gateway through `localhost:30600`.

> However, note that kubernetes' port forwarder incurs substantial overhead and

Reviewer: ```suggestion
However, the Kubernetes port forwarder incurs substantial overhead and
```

### Configure MinIO

If you are not using the MinIO client, skip this section.

To install and configure MinIO, complete the following steps:

1. Install the MinIO client on your platform as
described on the [MinIO download page](https://min.io/download#/macos).

> 2. Verify that MinIO components are successfully installed by running

Author: This way the markdown source is more readable, which is how I'll often go through the documents. Happy to change it back though, if we want to use lazy lists as a convention.

> 2. Verify that MinIO components are successfully installed by running

Reviewer: Not a blocker, but would appreciate if you change it back =)
